File: cases/taint/csharp/branch-join-negative/BranchJoinNegative.cs
namespace DataFlowBench;

static class BranchJoinNegative
{
    static int dfb_source() // DFB-SOURCE: branch-join-input
    {
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: branch-join-sink

    static void Run(bool overwrite)
    {
        int value = dfb_source();
        if (overwrite)
        {
            value = 0;
        }
        else
        {
            value = 0;
        }
        // DFB-WITNESS: branch-join-value
        dfb_sink(value);
    }
}

File: cases/taint/go/branch-join-negative/branch_join_negative.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: branch-join-input
	return 1
}

func dfb_sink(value int) {} // DFB-SINK: branch-join-sink

func run(overwrite bool) {
	value := dfb_source()
	if overwrite {
		value = 0
	} else {
		value = 0
	}
	// DFB-WITNESS: branch-join-value
	dfb_sink(value)
}
